distributed/src/main/scala/asobu/distributed/service/Controller.scala

```diff
@@ -0,0 +1,46 @@
+package asobu.distributed.service
+
+import akka.actor.ActorSystem
+import asobu.distributed.gateway.Endpoint.Prefix
+import asobu.distributed.EndpointDefinition
+import play.routes.compiler.{HandlerCall, Route}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+ * Bare bone controller without syntax and facilitators, use [[DistributedController]] in normal cases
+ */
+trait Controller {
+  /**
+   * Used to get route file "$name.route"
+   *
+   * @return
+   */
+  def name: String = getClass.getSimpleName.stripSuffix("$")
+  def prefix = Prefix("/")
+
+  lazy val routes: List[Route] = EndpointDefinitionParser.parseResource(prefix, s"$name.routes") match {
+    case Right(rs) ⇒ rs
+    case Left(err) ⇒ throw RoutesParsingException(err.map(_.toString).mkString(". "))
+  }
+
+  private def findRoute(action: Action): Route = routes.find { r ⇒
+    val HandlerCall(packageName, controllerName, _, method, _) = r.call
+    action.name == packageName + "." + controllerName + "." + method
+  }.getOrElse(throw new Exception(s"Cannot find route for action ${action.name}"))
+
+  def addAction(action: Action)(
+    implicit
+    registryClient: EndpointsRegistryClient,
+    ec: ExecutionContext,
+    sys: ActorSystem
+  ): Future[EndpointDefinition] = {
+    val epd: EndpointDefinition = toEndPointDefinition(action)
+    registryClient.add(epd).map(_ ⇒ epd)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Non ascii characters are not allowed](https://www.codacy.com/app/kailuo-wang/asobu/file/2251247463/issues/source?bid=3037259&fileBranchId=3178877#l39)